src/promptflow-devkit/promptflow/_sdk/operations/_flow_operations.py

```diff
@@ -1032,43 +1033,85 @@ def _infer_signature(self, entry: Callable, keep_entry: bool = False, validate:
             del flow_meta["entry"]
         return flow_meta, source_path.parent
 
+    @monitor_operation(activity_name="pf.flows.infer_signature", activity_type=ActivityType.PUBLICAPI)
+    def infer_signature(self, entry: Callable) -> dict:
+        """Extract signature for a callable class or a function. Signature indicates the ports of a flex flow using
+        the callable as entry.
+
+        If entry is a callable function, the signature includes inputs and outputs. For example:
+        def my_func(a: int, b: str) -> int:
+            pass
+        Signature:
+        {
+            "inputs": {
+                "a": {
+                    "type": "int",
```

Comment:
what's the cost of supporting below?
int -> integer 
float -> number

Reply:
UX/extension should recognize integer and number for flex flow if we do this mapping in flow meta generation given flex flow is still private;
If we want to apply this to tools.json, too:
1. UX/extension should recognize both;
2. customer need to upgrade their extension if they're using later version of promptflow